docs/samples/Microsoft.ML.Samples/Dynamic/Transforms/CalculateFeatureContribution.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ML;
+
+namespace Samples.Dynamic
+{
+    public static class CalculateFeatureContribution
+    {
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging, 
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed: 1);
+
+            // Create a small dataset.
+            var samples = Data.GenerateData();
+
+            // Convert training data to IDataView.
+            var data = mlContext.Data.LoadFromEnumerable(samples);
+
+            // Create a pipeline to concatenate the features into a feature vector and normalize it.
+            var transformPipeline = mlContext.Transforms.Concatenate("Features", Data.FeatureColumns)
+                .Append(mlContext.Transforms.NormalizeMeanVariance("Features"));
+
+            // Fit the pipeline.
+            var transformer = transformPipeline.Fit(data);
+
+            // Transform the data.
+            var transformedData = transformer.Transform(data);
+
+            // Define a linear trainer.
+            var linearTrainer = mlContext.Regression.Trainers.Ols();
```

Comment:
>mlContext.Regression.Trainers.Ols() [](start = 32, length = 35)

Can we not combined it with the pipeline above? Is there any specific reason for doing so?

Reply:
I'd like to compute the feature contributions on the transformed data, so I separate this out into two steps.

---
In reply to: [273270793](https://github.com/dotnet/machinelearning/pull/3241#discussion_r273270793) [](ancestors = 273270793)